build/xtask/src/auxflash.rs

```diff
@@ -34,20 +45,26 @@ fn pack_blob(blob: &AuxFlashBlob) -> Result<tlvc_text::Piece> {
     } else {
         data
     };
+    let blob_checksum = Sha3_256::digest(&data);
+
     let tag: [u8; 4] = blob.tag.as_bytes().try_into().unwrap();
-    Ok(tlvc_text::Piece::Chunk(
+    let piece = tlvc_text::Piece::Chunk(
         tlvc_text::Tag::new(tag),
         vec![tlvc_text::Piece::Bytes(data)],
-    ))
+    );
+    Ok((piece, blob_checksum.into()))
 }
 
 /// Constructs an auxiliary flash image, based on RFD 311
 ///
 /// Returns the checksum and the raw data to be saved
-pub fn build_auxflash(aux: &AuxFlash) -> Result<([u8; 32], Vec<u8>)> {
+pub fn build_auxflash(aux: &AuxFlash) -> Result<AuxFlashData> {
     let mut auxi = vec![];
+    let mut blob_checksums = BTreeMap::new();
     for f in &aux.blobs {
-        auxi.push(pack_blob(f)?);
+        let (piece, checksum) = pack_blob(f)?;
+        auxi.push(piece);
+        blob_checksums.insert(f.tag.clone(), checksum);
```

Comment:
Tiny nit - is `.clone()` necessary? I would expect tags to be `Copy`, I think

Reply:
They're `String` to make `#[derive(Deserialize)]` work, so `.clone()` is needed